bootstrap/internal/cloudinit/cloudinit.go

```diff
@@ -159,3 +169,41 @@ func cleanupAdditionalCloudInit(cloudInitData string) (string, error) {
 
 	return res, nil
 }
+
+func cleanupArbitraryData(arbitraryData map[string]string) error {
+	if arbitraryData == nil {
+		return nil
+	}
+
+	m := make(map[string]interface{})
+
+	kind := "arbitrary_prepare"
+	tm := template.New(kind).Funcs(defaultTemplateFuncMap)
+
+	if _, err := tm.Parse(arbitraryTemplate); err != nil {
+		return errors.Wrap(err, "failed to parse arbitrary keys template")
+	}
+
+	t, err := tm.Parse(`{{template "arbitrary" .AdditionalArbitraryData}}`)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse arbitrary template")
+	}
+
+	var out bytes.Buffer
+	if err := t.Execute(&out, BaseUserData{
+		AdditionalArbitraryData: arbitraryData,
+	}); err != nil {
+		return errors.Wrapf(err, "failed to generate %s template", kind)
+	}
+
+	if err := yaml.Unmarshal(out.Bytes(), m); err != nil {
+		return fmt.Errorf("failed to unmarshal arbitrary cloud-init data: %w, please check if you put valid yaml data: %s", err, out.String())
+	}
+
+	// Remove ignored fields from the map
+	for _, field := range ignoredCloudInitFields {
```

Comment:
We should do this before we execute the template so that fields are removed before.